android/engine/src/main/java/org/smartregister/fhircore/engine/util/SecureSharedPreference.kt

```diff
@@ -67,9 +68,20 @@ class SecureSharedPreference @Inject constructor(@ApplicationContext val context
       ?.decodeJson<AuthCredentials>()
   }
 
+  fun saveSessionPin(pin: String) {
+    secureSharedPreferences.edit { putString(KEY_SESSION_PIN, pin) }
+  }
+
+  fun retrieveSessionPin() = secureSharedPreferences.getString(KEY_SESSION_PIN, null)
+
+  fun deleteSessionPin() {
+    secureSharedPreferences.edit { remove(KEY_SESSION_PIN) }
+  }
+
```

Comment:
Can this be made generic to access other preferences? So for saving you can pass the KEY & VALUE to save. 
Delete passed the KEY then delete is done for that key. Retrieve just does the getting of the KEY given

Reply:
@maimoonak Do you think this is realistic?

Reply:
The class actually is designed on same pattern https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/util/SecureSharedPreference.kt . For credentials we are hiding the complexity for consumer class. For pin its straight forward for now but we can think if there can be any other sub task for this... We can create a codehealth ticket to address the design of this class and have discussion with Elle and others in call. We can refactor it when address that issue.. @AbdulWahabMemon plz create a ticket for this

Reply:
I have raised new task https://github.com/opensrp/fhircore/issues/1053 for the required discussion as mentioned above.